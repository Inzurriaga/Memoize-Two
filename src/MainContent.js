import React from "react"
import HomePage from "./Homepage"
import TrainPage from "./TrainPage"

export default function MainContent(props){
        return(
            <div>
                {
                    props.page === "home" ?
                  <HomePage users={props.users}
                            user={props.user}/> :
                  <TrainPage users={props.users}
                            user={props.user}
                            Questiontoggle={props.Questiontoggle}
                            toggleToQuestion={props.toggleToQuestion}
                            selectedQuestion={props.selectedQuestion}/>  
                }
            </div>
        )
}