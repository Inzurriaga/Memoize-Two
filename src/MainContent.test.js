import React from 'react';
import { shallow } from 'enzyme';
import MainContent from "./MainContent"

const mockPage = "home";
const mockUser = {"name":"gabriel","password":"gabriel","completed":[1,2,5]};
const mockUsers = [{"name":"duy","password":"duy","completed":[1,2,5]},{"name":"gabriel","password":"gabriel","completed":[1,2,5]},{"name":"isaac","password":"isaac","completed":[1,2]}];
const mockToggle = true;
const mockToggleFunction = jest.fn();
const mockSelectedQuestion = { title: "number return 5", type: "number", instructions: "return the number 5", answer: 5, startup: "function number() {\n//code here \n} \n\n number()", id: 1 };

describe("MainContent", ()=> {
  let wrapper;
  beforeEach(() => {
      wrapper = shallow(
        <MainContent page={mockPage}
                    user={mockUser}
                    users={mockUsers}
                    Questiontoggle={mockToggle}
                    toggleToQuestion={mockToggleFunction}
                    selectedQuestion={mockSelectedQuestion}/>)
  })

  it("should have a snapshot", () => {
      expect(wrapper).toMatchSnapshot()
  })
})