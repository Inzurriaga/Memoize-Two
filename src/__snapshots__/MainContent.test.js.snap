// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent should have a snapshot 1`] = `
<div>
  <HomePage
    user={
      Object {
        "completed": Array [
          1,
          2,
          5,
        ],
        "name": "gabriel",
        "password": "gabriel",
      }
    }
    users={
      Array [
        Object {
          "completed": Array [
            1,
            2,
            5,
          ],
          "name": "duy",
          "password": "duy",
        },
        Object {
          "completed": Array [
            1,
            2,
            5,
          ],
          "name": "gabriel",
          "password": "gabriel",
        },
        Object {
          "completed": Array [
            1,
            2,
          ],
          "name": "isaac",
          "password": "isaac",
        },
      ]
    }
  />
</div>
`;
